core/src/main/scala/akka/persistence/cassandra/snapshot/CassandraSnapshotStore.scala

```diff
@@ -111,18 +97,26 @@ class CassandraSnapshotStore(cfg: Config, cfgPath: String)
   override def loadAsync(
       persistenceId: String,
       criteria: SnapshotSelectionCriteria): Future[Option[SelectedSnapshot]] = {
+    log.debug("loadAsync [{}] [{}]", persistenceId, criteria)
     // The normal case is that timestamp is not specified (Long.MaxValue) in the criteria and then we can
     // use a select stmt with LIMIT if maxLoadAttempts, otherwise the result is iterated and
     // non-matching timestamps are discarded.
     val snapshotMetaPs =
       if (criteria.maxTimestamp == Long.MaxValue)
         preparedSelectSnapshotMetadataWithMaxLoadAttemptsLimit
       else preparedSelectSnapshotMetadata
-    for {
+    val result = for {
       p <- snapshotMetaPs
       mds <- metadata(p, persistenceId, criteria, someMaxLoadAttempts)
       res <- loadNAsync(mds)
     } yield res
+
+    result.onComplete { result =>
+      log.debug("snapshot load complete {}", result)
```

Comment:
Seems weird here, maybe consider just have it in the for comp's `yield`

Reply:
Removed, just added when nothing worked with the new driver